### PR TITLE
Updating Tuning Performance doc based on APIM 3.0.0

### PR DESCRIPTION
**Pull request by chamilaadhi**
Move content of this doc https://docs.wso2.com/display/AM260/Tuning+Performance to http://localhost:8000/InstallAndSetup/PerfromanceTuningAndTestResults/tuning-performance/

### CODE REVIEW

File: en/docs/InstallAndSetup/PerfromanceTuningAndTestResults/tuning-performance.md [hunk 1]
```diff
@@ -1,68 +1,68 @@
...
+This section describes some recommended performance tuning configurations to optimize the WSO2 API Manager. It assumes that you have set up the API Manager on Unix/Linux, which is recommended for a production deployment. We also recommend [a distributed API Manager setup](https://apim.docs.wso2.com/en/latest/InstallAndSetup/DeployingWSO2APIManager/deployment-patterns/) for most production systems. Out of all components of an API Manager distributed setup, the API Gateway is the most critical, because it handles all inbound calls to APIs. Therefore, we recommend you to have at least a 2-node cluster of API Gateways in a distributed setup.
```
Comment (tgtshanika) [inline en/docs/InstallAndSetup/PerfromanceTuningAndTestResults/tuning-performance.md]: Use {{base_path}}/InstallAndSetup/DeployingWSO2APIManager/deployment-patterns/

Comment (chamilaadhi) [reply en/docs/InstallAndSetup/PerfromanceTuningAndTestResults/tuning-performance.md]: fixed

File: en/docs/InstallAndSetup/PerfromanceTuningAndTestResults/tuning-performance.md [hunk 2]
```diff
@@ -262,13 +236,18 @@ Some general APIM-level recommendations are listed below:
...
-<div class="admonition danger">
-     <p class="admonition-title">Not supported</p>
-     <p>
-     Cannot configure using deployment.toml</p>
-     </div> 
+<td><p>Defines the maximum time that a callback waits in the Gateway for a response from the backend. If no response is received within this time, the Gateway drops the message and clears out the callback. This is a global level parameter that affects all the endpoints configured in the Gateway.</p>
+<p>The global timeout is defined in the <code>                    &lt;API-M_HOME&gt;/repository/conf/deployment.toml                   </code> file. The recommended value is 120000 ms.</p>
+<div class="code panel pdl" style="border-width: 1px;">
+<div class="codeContent panelContent pdl">
+<pre class="java" data-syntaxhighlighter-params="brush: java; gutter: false; theme: Confluence" data-theme="Confluence" style="brush: java; gutter: false; theme: Confluence"><code>
```
Comment (Mariangela) [inline en/docs/InstallAndSetup/PerfromanceTuningAndTestResults/tuning-performance.md]: Are these <div> and <pre> tags valid? The fact that it states Confluence means that this is some formatting that has got copied over from Confluence during the Confluence to Git doc migration process

File: en/docs/InstallAndSetup/PerfromanceTuningAndTestResults/tuning-performance.md [hunk 3]
```diff
@@ -81,41 +81,54 @@ In a clustered environment, the entity expansion limit has no dependency on the
...
 
 ##### Timeout configurations for an API call
 
 The following diagram shows the communication/network paths that occur when an API is called. The timeout configurations for each network call are explained below.
-![]({{base_path}}/assets/attachments/103333648/103333650.png)
+
+![](../../../../assets/attachments/103333648/103333650.png)
+
+!!! info
+    Gateway to Keymanager network call to validate token only happens with the OAuth token. This network call does not happen for JWT tokens . From WSO2 API Manager 3.0.0 onwards, JWT tokens are the default token type for applications. Since JWT tokens are self-contained access tokens, Keymanager is not needed to validate the token. The token is validated from the gateway.
```
Comment (Mariangela) [inline en/docs/InstallAndSetup/PerfromanceTuningAndTestResults/tuning-performance.md]: - Keymanager should be Key Manager
- There is an extra space after JWT tokens and the period (.)

I reworded the sentence based on the above two reasons and also to improve the grammar. Therefore, shall we reword this to the following:

The Gateway to Key Manager network call to validate the token only happens with the OAuth token. This network call does not happen for JWT tokens. From WSO2 API Manager 3.0.0 onwards, JWT tokens are the default token type for applications. As JWT tokens are self-contained access tokens, the Key Manager is not needed to validate the token and the token is validated from the Gateway.

File: en/docs/InstallAndSetup/PerfromanceTuningAndTestResults/tuning-performance.md [hunk 4]
```diff
@@ -81,41 +81,54 @@ In a clustered environment, the entity expansion limit has no dependency on the
...
+        <parameter name="SO_TIMEOUT">60000</parameter>
+        <parameter name="CONNECTION_TIMEOUT">60000</parameter>
+    </transportSender>
     ```
 
     If the Key Manager caching is enabled, the calls between the API Gateway and Key Manager are cached. As a result, the Key Manager is not invoked for each API call.
 
 -   **Client call API Gateway + API Gateway call Backend**
-    For backend communication, the API Manager uses PassThrough transport. This is configured in the `<API-M_HOME>/repository/conf/passthru-http.properties` file. For more information, see [Configuring passthru-http.properties](https://docs.wso2.com/display/EI620/Configuring+passthru-http.properties) in the EI documentation.
+    For backend communication, the API Manager uses PassThrough transport. This is configured in the `<API-M_HOME>/repository/conf/deployment.toml` file. For more information, see [Configuring passthru properties](https://docs.wso2.com/display/EI620/Configuring+passthru-http.properties) in the EI documentation. To configure socket timeout value, add following to the `deployment.toml`
```
Comment (Mariangela) [inline en/docs/InstallAndSetup/PerfromanceTuningAndTestResults/tuning-performance.md]: Why are we pointing to EI 6.2.0 when EI 6.5.0 is the lastest EI version that is compatible with APIM 3.0.0? The link needs to be updated.

File: en/docs/InstallAndSetup/PerfromanceTuningAndTestResults/tuning-performance.md [hunk 5]
```diff
@@ -270,42 +261,48 @@ Make the number of threads equal to the number of processor cores.</td>
...
-<p>Note that when the endpoint is suspended, the default action is defined here as invoking the fault sequence.</p>
-</p>
-<p><img src="attachments/50518314/80711443.png" alt="timeout.png" width="734" /></p></li>
-<li><p>Click <strong>Save</strong> and re-publish the API.</p></li>
+
+<div class="admonition note">
+     <p class="admonition-title">Note</p>
+     <p>Note that when the endpoint is suspended, the default action is defined here as invoking the fault sequence.</p>
+</div> 
+<p><img src="../../../../assets/attachments/103333648/103333649.png" alt="timeout.png" width="734" /></p></li>
```
Comment (Mariangela) [inline en/docs/InstallAndSetup/PerfromanceTuningAndTestResults/tuning-performance.md]: Image path needs to be updated and the image needs to be moved out of the assets/attachments/ folder as mentioned before.

File: en/docs/InstallAndSetup/PerfromanceTuningAndTestResults/tuning-performance.md [hunk 6]
```diff
@@ -368,29 +370,38 @@ Make the number of threads equal to the number of processor cores.</td>
...
 <div class="code panel pdl" style="border-width: 1px;">
 <div class="codeContent panelContent pdl">
-<pre class="java" data-syntaxhighlighter-params="brush: java; gutter: false; theme: Confluence" data-theme="Confluence" style="brush: java; gutter: false; theme: Confluence"><code>maxThreads=&quot;750&quot; 
-minSpareThreads=&quot;150&quot; 
-disableUploadTimeout=&quot;false&quot; 
-enableLookups=&quot;false&quot; 
-connectionUploadTimeout=&quot;120000&quot; 
-maxKeepAliveRequests=&quot;600&quot; 
-acceptCount=&quot;600&quot; </code></pre>
+<pre class="java" data-syntaxhighlighter-params="brush: java; gutter: false; theme: Confluence" data-theme="Confluence" style="brush: java; gutter: false; theme: Confluence"><code>[transport.https.properties]
```
Comment (Mariangela) [inline en/docs/InstallAndSetup/PerfromanceTuningAndTestResults/tuning-performance.md]: Check the <pre> formatting it looks like something based on Confluence.

File: en/docs/InstallAndSetup/PerfromanceTuningAndTestResults/tuning-performance.md [hunk 7]
```diff
@@ -401,27 +412,27 @@ acceptCount=&quot;600&quot; </code></pre>
 
 ##### Registry indexing configurations
 
-The registry indexing process is only required to be run on the API Publisher and API Store nodes. To disable the indexing process from running on the other nodes (Gateways and Key Managers), you need to set the `<wso2registry><indexingConfiguration><startIndexing>` element to false in the `<API-M_HOME>/repository/conf/registry.xml` file of the relevant nodes.
+The registry indexing process is only required to be run on the API Publisher and API Store nodes. To disable the indexing process from running on the other nodes (Gateways and Key Managers), you need to following in the `<API-M_HOME>/repository/conf/deployment.toml` file.
```
Comment (Mariangela) [inline en/docs/InstallAndSetup/PerfromanceTuningAndTestResults/tuning-performance.md]: Shouldn't "API Store" be "Developer Portal"?

File: en/docs/InstallAndSetup/PerfromanceTuningAndTestResults/tuning-performance.md (hunk 4 above)
Comment (chamilaadhi) [reply en/docs/InstallAndSetup/PerfromanceTuningAndTestResults/tuning-performance.md]: fixed

File: en/docs/InstallAndSetup/PerfromanceTuningAndTestResults/tuning-performance.md (hunk 3 above)
Comment (chamilaadhi) [reply en/docs/InstallAndSetup/PerfromanceTuningAndTestResults/tuning-performance.md]: updated

File: en/docs/InstallAndSetup/PerfromanceTuningAndTestResults/tuning-performance.md (hunk 5 above)
Comment (chamilaadhi) [reply en/docs/InstallAndSetup/PerfromanceTuningAndTestResults/tuning-performance.md]: fixed

File: en/docs/InstallAndSetup/PerfromanceTuningAndTestResults/tuning-performance.md [hunk 8]
```diff
@@ -265,7 +266,7 @@ Some general APIM-level recommendations are listed below:
      <p class="admonition-title">Note</p>
      <p>Note that when the endpoint is suspended, the default action is defined here as invoking the fault sequence.</p>
 </div> 
-<p><img src="../../../../assets/attachments/103333648/103333649.png" alt="timeout.png" width="734" /></p></li>
+<p><img src="../../../../../assets/img/setup-and-install/timeout-configuration.png" alt="timeout-configuration.png" width="734" /></p></li>
```
Comment (Mariangela) [inline en/docs/InstallAndSetup/PerfromanceTuningAndTestResults/tuning-performance.md]: ../ should be replaced with {{base_path}}
